geps/gep-1867.md

```diff
@@ -0,0 +1,134 @@
+
+[Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment) offers a hierarchy of defaults and overrides, allowing attachment to GatewayClass and Gateway.
+This is similar to our needs here, but representing infrastructure configuration as a "Policy" is a bit problematic, and the existing mechanisms have no hierarchy.
+
+In core Kubernetes, Pods declare their requirements (for example, CPU requests) inline in the Pod resource; there is not a `ResourceClass` API that abstracts these further.
+These higher level abstractions are handled by layered APIs (whether this is a CRD, an admission webhook, CI/CD tooling, etc).
+This allows users the flexibility to easily configure things per-pod basis.
+If the infrastructure admin wants to impose defaults or requirements on this flexibility, they are able to do so (in fact, `LimitRanger` provides a built in mechanism to do so).
+
+### Dynamic Changes
+
+Currently, the spec recommends `GatewayClass` to be used as a *template*.
+Changes to it are not expected to change deployed `Gateway`s.
+
+This makes usage problematic in a declarative way.
+For example, if I wanted to represent a `version` field and change that to trigger an upgrade, I would need to create an entirely new `Gateway`.
+
+## API
+
+In order to address the concerns above, I propose a standard `infrastructure` API is added to `Gateway` and `GatewayClass`.
+Note the important part of this is the `Gateway` change; the `GatewayClass` aspect is mostly for consistency.
+
+The exact fields are out of scope for this GEP and will be handled by additional GEPs.
+Some example GEPs already depending on this are [GEP-1713](/geps/gep-1713.md), [GEP-1651](/geps/gep-1651.md), and [GEP-1762](/geps/gep-1762.md)
+
+The fields as defined below are, of course, not useful.
+This is intended as a basis for other PRs, not to provide value on its own.
+This GEP will remain in provisional until at least one field is ready to be promoted.
+
+```go
+type GatewaySpec struct {
+  // Infrastructure defines infrastructure level attributes about this Gateway instance.
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+
+type GatewayClassSpec struct {
+  // Infrastructure defines infrastructure level attributes for all Gateways in this class.
+  // A Gateway may provide configuration for the same values; as all fields in GatewayInfrastructure are implementation specific,
+  // the merging logic between these is as well. However, the GatewayClass is generally expected to be providing defaults
+  // rather than overrides.
+  Infrastructure GatewayClassInfrastructure `json:"infrastructure"`
+  // ...
+}
+
+type GatewayInfrastructure struct {
+  // ParametersRef provides a arbitrary implementation-specific configuration for
+  // fields not expressed directly in this struct.
+  // This follows the same semantics as GatewayClass's ParametersRef, but lives on the Gateway.
+  ParametersRef ParametersReference
```

Comment:
Can you copy the relevant spec from GWC here? I think we may need a bit more than that written to also clarify what this is intended to be used for. This could become a very generic Gateway extension mechanism for things that aren't exactly infra-related (timeouts, retries, auth, etc). I don't think that's intended here, but if we don't clearly define what is meant to be configured here, it may end up getting used in unintended ways.

https://github.com/kubernetes-sigs/gateway-api/blob/4738c6c6e981d4a0e38c8c6120dc0198e4128351/apis/v1beta1/gatewayclass_types.go#L83-L96